### PR TITLE
Screenshoter: fix screenshot folder

### PR DESCRIPTION
**Pull request by hius07**
Check for existance and create if absent.
Closes https://github.com/koreader/koreader/issues/12742.

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/12750)

### CODE REVIEW

**Review by NiLuJe — APPROVED** [review]

**Comment by NiLuJe** [comment]
Oops, skipped over the CI: `frontend/ui/widget/screenshoter.lua:103:39: shadowing upvalue argument caller_callback on line 49` ;).

**Comment by benoit-pierre** [comment]
*sigh*

Something changed on CircleCI's side, so we're trying to build with too many jobs again:
```
************ NINJA: ninja -j54 -l36 **********
************ MAKE: make -j54 -l36 **********
```

**Comment by hius07** [comment]
Dunno if it's related, a notice from them:

Notice: If you are using "when: always" in your workflow filters, update to "when: true" to avoid configuration failures.

**Comment by benoit-pierre** [comment]
> Dunno if it's related, a notice from them:
> 
> Notice: If you are using "when: always" in your workflow filters, update to "when: true" to avoid configuration failures.

Confusing and misleading, this does not apply to our case (Cf. https://circleci.com/changelog/dynamic-when-statements-breaking-change/):
> NOTE: When in Steps are not impacted by this change.

File: frontend/ui/widget/screenshoter.lua [hunk 1]
```diff
@@ -100,8 +100,8 @@ function Screenshoter:onScreenshot(screenshot_name, caller_callback)
         title = _("Screenshot saved to:") .. "\n\n" .. BD.filepath(screenshot_name) .. "\n",
         modal = true,
         buttons = buttons,
-        tap_close_callback = function(caller_callback)
-            if caller_callback then
+        tap_close_callback = function()
+            if type(caller_callback) == "function" then
```
Comment (NiLuJe) [inline frontend/ui/widget/screenshoter.lua:104]: Err, can it be anything else?

**Pull request merged**